dags/examples/maxtext_sweep_gke_example_dag.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+An example DAG to launch a sweep of MaxText GKE XPK workloads
+sweeping over a range of per_device_batch_size values for a 16B model
+on 1xv4-128.
+"""
+
+import datetime
+from airflow import models
+from dags import test_owner
+from dags.vm_resource import TpuVersion, Zone, Project
+from dags.multipod.configs import maxtext_sweep_gke_config
+
+
+with models.DAG(
+    dag_id="maxtext_sweep_gke_example_dag",
+    schedule=None,
+    tags=["multipod_team", "maxtext"],
+    start_date=datetime.datetime(2024, 1, 10),
+    catchup=False,
+) as dag:
+  # MaxText set up and run commands
+  base_run_model_cmds = [
+      "python3 MaxText/train.py MaxText/configs/base.yml base_output_directory=gs://maxtext-experiments-multipod/ dataset_path=gs://max-datasets-rogue enable_checkpointing=false global_parameter_scale=16 steps=10",
+  ]
+
+  # Get list of MaxText GKE XPK jobs
+  maxtext_sweep_gke_test = maxtext_sweep_gke_config.get_maxtext_sweep_gke_config(
+      test_owner=test_owner.RAYMOND_Z,
+      project_name=Project.TPU_PROD_ENV_MULTIPOD.value,
+      cluster_name="v4-bodaborg",
+      tpu_zone=Zone.US_CENTRAL2_B.value,
+      time_out_in_min=60,
+      tpu_version=TpuVersion.V4,
+      tpu_cores=128,
+      num_slices=[1],
+      docker_image=f"gcr.io/{Project.TPU_PROD_ENV_MULTIPOD.value}/raymondzou_runner",
```

Comment:
@raymondzouu I've been experimenting with having a `docker build` step as part of the DAG to keep the image fresh for each run: http://shortn/_RArxnOOBkp

If you think this would be useful for these tests, we can bundle some of your common setup commands into a Dockerfile in this repo and rebuild the image before each config sweep. Let me know what you think, I can land the docker build step separately from my WIP pytorch tests.

This is really only applicable to the XPK tests though, so not blocking.

Reply:
@jonb377 I think a docker build step would be really helpful for nightly XLML tests with XPK since it would need a new nightly build in the image every time it runs. For this case, a benefit of the user supplying there own docker image is that they can have there own maxtext/dependency code changes in the image. Would the new docker build step be able to support this?

Reply:
As I have it now, the docker build runs through cloud build, so it can operate directly on a `Dockerfile` or `cloudbuild.yaml` in the repo. I've only worked with the former use case so far, but using a `cloudbuild.yaml` should be flexible enough to build a docker image from a `Dockerfile` in another repo or run a custom build script.

It sounds like the use case you're targeting here is more for the user to build their own image with local changes to supply to the parameter sweep, is that right? If so, I would say that doesn't sound like a good fit for building the image as part of the DAG, so what you have now is fine. Let me know what you think!

Reply:
Yep that is right, for this case I think it is iimportant to be able to supply local changes. Sounds good, I will leave it as is. Thank you for bringing this up!